src/Bridges/bridge_optimizer.jl

```diff
@@ -546,13 +546,26 @@ function _delete_variables_in_variables_constraints(
 end
 
 function MOI.delete(b::AbstractBridgeOptimizer, vis::Vector{MOI.VariableIndex})
-    F = MOI.get(b, MOI.ObjectiveFunctionType())
-    if is_objective_bridged(b) && F == MOI.VectorOfVariables
-        f = MOI.get(b, MOI.ObjectiveFunction{MOI.VectorOfVariables}())
-        discard = Base.Fix2(in, vis)
-        if any(discard, f.variables)
-            g = MOI.VectorOfVariables(filter(!discard, f.variables))
-            MOI.set(b, MOI.ObjectiveFunction{MOI.VectorOfVariables}(), g)
+    if is_objective_bridged(b)
```

Comment:
You can use `&&`, the second clause won't be executed if the first one is `false` and I reiterate https://github.com/jump-dev/MathOptInterface.jl/pull/2147#issuecomment-1512526018, having a complicated logic duplicated like that, you may end up trying to decipher twice the same code